### DISCUSSION TITLE
PrivateCookieJar that doesn't depend on app state?

### DISCUSSION THREAD

**Discussion by jakswa** [tokio-rs/axum#2533] [General]
Hello I'm trying to add PrivateCookieJar and reading the doc page (https://docs.rs/axum-extra/latest/axum_extra/extract/cookie/struct.PrivateCookieJar.html) but don't understand how to feed the key in from env or some other source besides app state.

Does a private cookie jar's key _have_ to come from the app state?

The context here is that I'm testing out https://loco.rs and am not able to implement `from` on its `AppContext` that I can see.

**Comment by davidpdrsn** [comment]
Yes the key must come from the state. So you have to put the env car in your state.

**Reply by jakswa** [reply]
Thanks, makes sense. Just making sure before I give tower-cookies a try. Wish me luck 🫡

**Reply by davidpdrsn** [reply]
Godspeed
